src/Material/Footer.elm

```diff
@@ -0,0 +1,339 @@
+module Material.Footer
+  exposing
+    ( Type(..)
+    , Property
+    , Content(..)
+    , Footer, Element
+    , mini, mega, footer
+    , left, right, top, bottom, middle
+    , wrap
+    , logo, text, socialButton
+    , href, link, onClick
+    , dropdown, heading
+    , links, linkItem
+    )
+
+{-| From the [Material Design Lite documentation](https://getmdl.io/components/index.html#layout-section/footer):
+
+> The Material Design Lite (MDL) footer component is a comprehensive container
+> intended to present a substantial amount of related content in a visually
+> attractive and logically intuitive area. Although it is called "footer", it
+> may be placed at any appropriate location on a device screen, either before or
+> after other content.
+>
+> An MDL footer component takes two basic forms: mega-footer and mini-footer. As
+> the names imply, mega-footers contain more (and more complex) content than
+> mini-footers. A mega-footer presents multiple sections of content separated by
+> horizontal rules, while a mini-footer presents a single section of content. Both
+> footer forms have their own internal structures, including required and optional
+> elements, and typically include both informational and clickable content, such
+> as links.
+>
+> Footers, as represented by this component, are a fairly new feature in user
+> interfaces, and allow users to view discrete blocks of content in a coherent and
+> consistently organized way. Their design and use is an important factor in the
+> overall user experience.
+
+See also the
+[Material Design Specification](https://material.google.com/layout/structure.html).
+
+Refer to [this site](http://debois.github.io/elm-mdl#/footer)
+for a live demo.
+
+# Types
+
+@docs Type
+@docs Content, Footer, Element
+@docs Property
+
+# Helpers
+
+@docs wrap
+@docs link, onClick, href
+
+# Appearance
+
+@docs footer, mini, mega
+
+# Sections
+
+@docs left, right, top, bottom, middle
+
+# Content
+
+@docs links, logo, text, socialButton, dropdown, heading, linkItem
+
+-}
+
+import Html exposing (..)
+import Html.Attributes as Html
+import Html.Events as Events
+import Material.Options as Options exposing (Style, cs)
+
+import String
+import Regex
+import Material.Options.Internal as Internal exposing (attribute)
+
+
+{-| The type of the footer
+-}
+type Type
+  = Mini
+  | Mega
+
+
+prefix : Type -> String
+prefix tp =
+  case tp of
+    Mini -> "mdl-mini-footer"
+    Mega -> "mdl-mega-footer"
+
+separator : String
+separator = "__"
+
+{-| TODO
+-}
+type FooterProperty = FooterProperty
+
+{-| TODO
+-}
+type alias Property m =
+  Options.Property FooterProperty m
+
+{-| TODO
+-}
+type Content a
```

Comment:
I think what I had some problems with was trying to apply the prefixes based on the Type of the footer, which is not known until the call to `footer` so the actual `Html m` nodes should not be actually created until that point because currently there is no way to access properties of an existing node.

I was entertaining some thoughts on some sort of `lazy DOM` that would basically wrap existing `Html m` nodes but not create them until explicitly called, this would allow you to insert properties and "modify" them in cases like this.

In this case I think you could do without the `Content m` but the api would would need to be more explicit e.g. `miniLeft` that would apply the specific class